bcipy/helpers/task.py

```diff
@@ -174,6 +175,78 @@ def get_data_for_decision(inquiry_timing,
     return raw_data, triggers
 
 
+def get_device_data_for_decision(
+        inquiry_timing: List[Tuple],
+        daq: ClientManager,
+        offset: float = 0.0,
+        prestim: float = 0.0,
+        poststim: float = 0.0) -> Dict[ContentType, np.ndarray]:
+    """Queries the acquisition client manager for a slice of data from each
+    device and processes the resulting raw data into a form that can be passed
+    to signal processing and classifiers.
+
+    Parameters
+    ----------
+    - inquiry_timing(list): list of tuples containing stimuli timing and labels.
+    - daq (ClientManager): bcipy data acquisition client manager
+    - offset (float): offset present in the system which should be accounted for when creating data for classification.
+        This is determined experimentally.
+    - prestim (float): length of data needed before the first sample to reshape and apply transformations
+    - poststim (float): length of data needed after the last sample in order to reshape and apply transformations
+
+    Returns
+    -------
+        a dict mapping device content_type to the data;
+            data has shape C x L where C is number of channels and L is the
+            signal length.
+    """
+    _, first_stim_time = inquiry_timing[0]
+    _, last_stim_time = inquiry_timing[-1]
+
+    # adjust for offsets
+    time1 = first_stim_time + offset - prestim
+    time2 = last_stim_time + offset
+
+    if time2 < time1:
+        raise InsufficientDataException(
+            f'Invalid data query [{time1}-{time2}] with parameters:'
+            f'[inquiry={inquiry_timing}, offset={offset}, prestim={prestim}, poststim={poststim}]'
+        )
+
+    data = daq.get_data_by_device(start=time1,
+                                  seconds=(time2 - time1 + poststim),
+                                  strict=True)
+    for content_type in data.keys():
+        # Take only the sensor data from raw data and transpose it.
+        # TODO: Does this apply to all content types? If not this step could be

```

Comment:
I think it would. All data streamed from LSL should be able to be cast to a np.array. One thing to note is that some data can have other values, though. For instance, eye trackers can have nan in the gaze stream, and I don't know if that casting will work.

Reply:
I will leave this code as is for now but I created a Pivotal Tracker ticket to revisit this when we are integrating the Eyetracker model: https://www.pivotaltracker.com/story/show/185711288.